get-live-images.sh

```diff
@@ -0,0 +1,249 @@
+[DEFAULT]
+api_url = ${IRONIC_BASE_URL}:6385
+inspection_callback_url = ${IRONIC_BASE_URL}:5050/v1/continue
+
+collect_lldp = True
+enable_vlan_interfaces = ${IRONIC_INSPECTOR_VLAN_INTERFACES}
+inspection_collectors = default,extra-hardware,logs
+inspection_dhcp_all_interfaces = True
+_EOF_
+
+	IRONIC_AGENT_CONF=$(cat /tmp/ironic-python-agent.conf|base64 -w 0)
+
+cat > ${IGNITION_FILE} <<_EOF_
+{
+  "ignition": { "version": "3.2.0" },
+  "passwd": {
+    "users": [
+      {
+        "name": "core",
+        "sshAuthorizedKeys": [
+          "${IRONIC_RAMDISK_SSH_KEY}"
+        ]
+      }
+    ]
+  },
+  "storage": {
+    "files": [{
+      "path": "/etc/ironic-python-agent.conf",
+      "mode": 644,
+      "contents": {"source": "data:;base64,${IRONIC_AGENT_CONF}"}
+    }]
+  },
+  "systemd": {
+    "units": [
+      {
+        "contents": "[Unit]\nDescription=Ironic Agent\nAfter=network-online.target\nWants=network-online.target\n\n[Service]\nTimeoutStartSec=0\nExecStartPre=/bin/podman pull ${IRONIC_AGENT_IMAGE} --tls-verify=false\nExecStart=/bin/podman run --privileged --network host --mount type=bind,src=/etc/ironic-python-agent.conf,dst=/etc/ironic-python-agent/ignition.conf --mount type=bind,src=/dev,dst=/dev --mount type=bind,src=/sys,dst=/sys --mount type=bind,src=/,dst=/mnt/coreos --name ironic-agent ironic-agent\n\n[Install]\nWantedBy=multi-user.target\n",
+        "enabled": true,
+        "name": "ironic-agent.service"
+      }
+    ]
+  }
+}
+_EOF_
+
+	# Embed the agent ignition
+	coreos-installer iso ignition embed -i "$IGNITION_FILE" -f "$ISO_FILE"
+}
+
+mkdir -p /shared/html/images /shared/tmp
+
+URLS=($(echo $RHCOS_IMAGE_URL | tr "," "\n"))
+
+# This name will be used in the ironic image to embed the agent ignition
+FFILENAME="ironic-python-agent"
+for URL in "${URLS[@]}"
+do
+	URL_STRIPPED=`echo $URL | cut -f 1 -d \?`
+	if [[ $URL_STRIPPED =~ live-initramfs ]]; then
+		INITRAMFS_FILENAME=$(basename $URL_STRIPPED)
+		INITRAMFS_FILENAME_CACHED=cached-${INITRAMFS_FILENAME}
+		TMPDIR=$(mktemp -d -p /shared/tmp)
+		trap "rm -fr $TMPDIR" EXIT
+		cd $TMPDIR
+
+		# We have a file in the cache that matches the one we want, use it
+		if [ -s "/shared/html/images/$INITRAMFS_FILENAME/$INITRAMFS_FILENAME_CACHED.md5sum" ]; then
+			echo "$INITRAMFS_FILENAME/$INITRAMFS_FILENAME_CACHED.md5sum found, contents:"
+			cat /shared/html/images/$INITRAMFS_FILENAME/$INITRAMFS_FILENAME_CACHED.md5sum
+		else
+			download_image "${INITRAMFS_FILENAME_CACHED}" "${URL_STRIPPED}"
+		fi
+		md5sum "$INITRAMFS_FILENAME_CACHED" | cut -f 1 -d " " > "$INITRAMFS_FILENAME_CACHED.md5sum"
+
+		if [ -s "${INITRAMFS_FILENAME_CACHED}.md5sum" ] ; then
+			cd /shared/html/images
+			chmod 755 $TMPDIR
+			mv $TMPDIR $INITRAMFS_FILENAME
+			ln -sf "$INITRAMFS_FILENAME/$INITRAMFS_FILENAME_CACHED" "$FFILENAME.initramfs"
+			ln -sf "$INITRAMFS_FILENAME/$INITRAMFS_FILENAME_CACHED.md5sum" "$FFILENAME.initramfs.md5sum"
+		fi
+	elif [[ $URL_STRIPPED =~ live-kernel ]]; then
```

Comment:
The 4 parts are nearly identical, maybe extract a common helper?

Reply:
I agree. I separated out the downloading part and it makes sense to simplify further.